111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,239 @@
+Kaniko, once the image is build and pushed, is able to return the full name of the image together with the SHA to particular path.
+That is done using the `--image-name-with-digest-file` option of the Kaniko executor.
+Unfortunately, Buildah doesn't have such option in case that you want to build and push the image in one command (that is done using the `buildah build` command, in order to push directly, you need to prefix the image with `docker://`).
+Buildah has option to store the SHA to specify file on path using `--digestfile` option - but only when `push` command is used.
+It will just output the SHA, not the full image, so we need to build it from the SHA and built image and output it to `/dev/termination-log` in order to have it inside the message of the completed Pod.
+
+We will use following commands:
+
+```shell
+buildah build --file=/dockerfile/Dockerfile --tag=<IMAGE> --storage-driver=vfs <ADDITIONAL_BUILD_OPTS>
+buildah push --storage-driver=vfs --digestfile=/tmp/digest <ADDITIONAL_PUSH_OPTS> <IMAGE>
+buildah images --digests --filter=digest=sha256:$(cat /tmp/digest) --format='{{.Name}}@{{.Digest}}' > /dev/termination-log
+```
+- `<IMAGE>` is placeholder for user desired name of the image (with registry, repository, and possibly tag)
+- `<ADDITIONAL_BUILD_OPTS>` is placeholder for user desired additional options for the `buildah build` command
+- `<ADDITIONAL_PUSH_OPTS>` is placeholder for user desired additional options for the `buildah push` command
+
+We need to take these three steps in order to get the SHA of the image together with correct repository.
+This will work in all scenarios, including when the user does not specify a tag and the default (`latest`) is used.
+
+Together with this feature, and upcoming v1 API, we will do following changes inside the `DockerOutput` model:
+
+1. we will deprecate `additionalKanikoOptions` field as the name of it is connected directly to Kaniko and more generic name can be used (this field will be removed as part of the v1 API).
+2. we will add two new fields - `additionalBuildOptions` and `additionalPushOptions` - to cover additional options for both phases
+
+The `additionalKanikoOptions` will still be used with Kaniko until the `UseConnectBuildWithBuildah` feature gate moves to GA or until it is removed in the CRD v1 API, whatever comes first.
+An automated warning about deprecation will be added inside the `.status` section of the `KafkaConnect` CR in case it is used as for all other deprecated fields.
+Instead of `additionalKanikoOptions`, users should use `additionalBuildOptions`, but not `additionalPushOptions`.
+We will not split the Kaniko's additional options to two groups, and we will keep everything configured in one.
+That's because Kaniko doesn't have the build and push phases split, but everything is done in one command.
+The difference between the build and push options in Kaniko is that the `-pull` suffix is applied for options used only for the build phase.
```

Comment:
I don't really get the meaning here. Seems like it needs more detials or should be deleted?

Reply:
I just wanted to mention that the options for Kaniko's `build` phase are suffixed with `-pull` for example. But I guess it doesn't make sense to have this sentence there. Let me delete it.